minion/libwrapper.cpp

```diff
@@ -194,6 +197,109 @@ void finaliseModel(CSPInstance& instance) {
 }
 
 
+/*********************************************************************/
+/*                    REXPORTING INLINE FUNCTIONS                    */
+/*********************************************************************/
+
+SearchOptions* newSearchOptions() {
+  return new SearchOptions();
+}
+
+SearchMethod* newSearchMethod() {
+  return new SearchMethod();
+}
+
+CSPInstance* newInstance() {
+  return new CSPInstance();
+}
+
+ConstraintBlob* newConstraintBlob(ConstraintType constraint_type) {
+  return new ConstraintBlob(lib_getConstraint(constraint_type));
+}
+
+SearchOrder* newSearchOrder(std::vector<Var>& vars,VarOrderEnum orderEnum, bool findOneSol) {
+  return new SearchOrder(vars,orderEnum,findOneSol);
+}
+
+void searchOptions_free(SearchOptions* searchOptions) {
+  delete searchOptions;
+}
+
+void searchMethod_free(SearchMethod* searchMethod) {
+  delete searchMethod;
+}
+
+void instance_free(CSPInstance* instance) {
+  delete instance;
+}
+
+void constraint_free(ConstraintBlob* constraint) {
+  delete constraint;
+}
+
+void searchOrder_free(SearchOrder* searchOrder) {
+  delete searchOrder;
+}
+
+Var getVarByName(CSPInstance& instance, char* name) {
+
+  return instance.vars.getSymbol(string(name));
+}
+
+void newVar_ffi(CSPInstance& instance, char* name, VariableType type, int bound1, int bound2) {
+  newVar(instance, string(name), type, std::vector<DomainInt>({bound1,bound2}));
+}
+
+
+void instance_addSearchOrder(CSPInstance& instance, SearchOrder& searchOrder) {
+  instance.searchOrder.push_back(searchOrder);
+}
+
+void instance_addConstraint(CSPInstance& instance, ConstraintBlob& constraint) {
+  instance.constraints.push_back(constraint);
+}
+
+void printMatrix_addVar(CSPInstance& instance, Var var) {
+  instance.print_matrix.push_back({var});
+}
+
+int printMatrix_getValue(int idx) {
+  // FIXME: does not work with minion debug mode
+  return globals->state_m->getPrintMatrix()[idx][0].assignedValue();
```

Comment:
Replace this with:  return checked_cast<int>(globals->state_m->getPrintMatrix()[idx][0].assignedValue());

checked_cast is a template I wrote (it's designed to look like a C++ cast, if you are familiar with them).

In debug mode, DomainInt does overflow checking (which C++ doesn't do for normal ints), and you need to explictly turn it back into an 'int'.